File: ServerHotel-WebManager/ServerHotel-WebManager/models/Review.js
const mongoose = require('mongoose');
const schema = mongoose.Schema;

const reviewSchema = new schema({

    service: {
        type: mongoose.Schema.Types.ObjectId, ref : 'Service',
        require: false
    },
    room: {
        type: mongoose.Schema.Types.ObjectId, ref : 'Room',
        require: false
    },
    user: {
        type: mongoose.Schema.Types.ObjectId, ref : 'User',
        require: false
    },
    rating: {
        type: Number,
        require: true
    },
    comment:{
        type: String,
        require: true
    }
});

const Review = mongoose.model('Review', reviewSchema);

module.exports = Review;

File: Hopa-master/Hopa-master/app/src/main/java/com/poly/hopa/models/Review.java
package com.poly.hopa.models;

import com.google.gson.annotations.SerializedName;

public class Review {
    @SerializedName("_id")
    private String reviewId;
    private Room room;
    private Service service;
    private User user;
    private float rating;
    private String comment;

    public String getReviewId() {
        return reviewId;
    }

    public void setReviewId(String reviewId) {
        this.reviewId = reviewId;
    }

    public Room getRoom() {
        return room;
    }

    public void setRoom(Room room) {
        this.room = room;
    }

    public Service getService() {
        return service;
    }

    public void setService(Service service) {
        this.service = service;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public float getRating() {
        return rating;
    }

    public void setRating(float rating) {
        this.rating = rating;
    }

    public String getComment() {
        return comment;
    }

    public void setComment(String comment) {
        this.comment = comment;
    }
}
